core/src/nodes/collection.rs

```diff
@@ -788,9 +818,9 @@ mod tests {
         let fut = future::empty::<_, Void>();
         let reach_id = cs.add_reach_attempt(fut, Handler::default());
         assert!(cs.interrupt(reach_id).is_ok());
-        assert!(cs.interrupt(reach_id).is_err());
+        assert_eq!(cs.interrupt(reach_id), Err(InterruptError::ReachAttemptNotFound))
```

Comment:
Can you check if using `assert_matches!(cs.interrupt(reach_id)), Err(InterruptError::ReachAttemptNotFound));` works? I think it would let you omit the `PartialEq` bound on the error enum.